ee/tabby-webserver/src/schema/auth.rs

```diff
@@ -354,6 +370,8 @@ pub trait AuthenticationService: Send + Sync {
         first: Option<usize>,
         last: Option<usize>,
     ) -> Result<Vec<InvitationNext>>;
+
+    async fn get_oauth_credential(&self, provider: OAuthProvider) -> Result<OAuthCredential>;
```

Comment:
instead of reading oauth here, I'll suggest just creating an function to actual wrap the authentication logic, e.g

```rust
fn github_auth(...) ->  std::result::Result<TokenAuthResponse, TokenAuthError> {
}
```